### PR TITLE
=peq #21724 #19174 unify naming of query methods, un-experimentalize Query

### LINKED ISSUE #21724: Rename EventsByTagQuery2 and EventEnvelope2
And replace EventsByTagQuery and EventEnvelope.

### PR DESCRIPTION
**Pull request by ktoso**
- fix "all" naming, 
  - we consistency use "eventsBy..." for infinite, 
  - and "currentEventsBy..." for finite streams now
- rename the "EventsByTagQuery2" classes (we now use the `Offset`)
- un experimentalize persistence query

Resolves #21724 #19174

### CODE REVIEW

**Comment by ktoso** [comment]
Don't review yet. WIP

**Comment by ktoso** [comment]
This is ready for review now.

File: akka-persistence-query/src/main/scala/akka/persistence/query/javadsl/PersistenceIdsQuery.scala [hunk 1]
```diff
@@ -20,6 +20,6 @@ trait AllPersistenceIdsQuery extends ReadJournal {
    * Corresponding query that is completed when it reaches the end of the currently
    * currently used `persistenceIds` is provided by [[CurrentPersistenceIdsQuery#currentPersistenceIds]].
    */
-  def allPersistenceIds(): Source[String, NotUsed]
+  def persistenceIds(): Source[String, NotUsed]
```
Comment (johanandren) [inline akka-persistence-query/src/main/scala/akka/persistence/query/javadsl/PersistenceIdsQuery.scala]: Are you sure about this change? Will break a lot of code...

File: akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/EventsByTagPublisher.scala [hunk 2]
```diff
@@ -122,6 +121,7 @@ private[akka] abstract class AbstractEventsByTagPublisher(
 /**
  * INTERNAL API
  */
+// FIXME needs a be rewritten as a GraphStage (since 2.5.0)
```
Comment (johanandren) [inline akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/EventsByTagPublisher.scala]: I concur

File: akka-persistence-query/src/main/scala/akka/persistence/query/javadsl/PersistenceIdsQuery.scala (hunk 1 above)
Comment (johanandren) [inline akka-persistence-query/src/main/scala/akka/persistence/query/javadsl/PersistenceIdsQuery.scala]: Ok

**Comment by johanandren** [comment]
Seems to be some docs not compiling anymore.

**Comment by ktoso** [comment]
Ouch docs, forgot to run tests there - thanks @johanandren

**Comment by ktoso** [comment]
Should pass now, please review.

File: akka-persistence-query/src/main/scala/akka/persistence/query/Offset.scala [hunk 3]
```diff
@@ -15,13 +15,13 @@ object Offset {
 
 }
 
-trait Offset
+class Offset
```
Comment (patriknw) [inline akka-persistence-query/src/main/scala/akka/persistence/query/Offset.scala]: why is that better? if so, it should be `abstract class`

File: akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst [hunk 4]
```diff
@@ -80,6 +80,38 @@ non-sharable journal or snapshot store. The proxy is available by setting ``akka
...
+For example, in some journals an offset is always a time, while in others it is a numeric offset (like a sequence id).
+
+Instead of the previous ``Long`` offset you can now use the provided ``Offset`` factories (and types):
+
+- ``akka.persistence.query.Offset.sequence(value: Long)``, 
+- ``akka.persistence.query.Offset.timeBasedUUID(value: UUID)``
+- and finally ``NoOffset`` if not offset should be used. 
+
+Journals are also free to provide their own specific ``Offset`` types. Consult your journal plugin's documentation for details.
+
```
Comment (patriknw) [inline akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst]: There is also this page that is listing experimental modules: http://doc.akka.io/docs/akka/2.4/experimental/index.html

**Review by patriknw — APPROVED** [review]
LGTM, with minor comments

File: akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst (hunk 4 above)
Comment (ktoso) [inline akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst]: Thanks!

Comment (ktoso) [inline akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst]: Removed from there.

**Comment by ktoso** [comment]
This is ready. Please review and merge at will.

**Comment by ktoso** [comment]
This is ready. Please review and merge at will. @johanandren

File: akka-persistence-query/src/main/scala/akka/persistence/query/Offset.scala (hunk 3 above)
Comment (patriknw) [inline akka-persistence-query/src/main/scala/akka/persistence/query/Offset.scala]: ah, that one, but `abstract class` should work?

Comment (ktoso) [inline akka-persistence-query/src/main/scala/akka/persistence/query/Offset.scala]: Yes, absolutely - making it abstract class then

**Comment by johanandren** [comment]
Mima is unhappy, and two unrelated test failures it seems.

File: project/MiMa.scala [hunk 5]
```diff
@@ -38,9 +39,20 @@ object MiMa extends AutoPlugin {
         "akka-http-testkit",
         "akka-stream-testkit"
       )
+      val akka2415NewArtifacts = Seq(
+        "akka-persistence-query"
+      )
```
Comment (ktoso) [inline project/MiMa.scala]: Ah of course. Yes should be empty (was now too :)). Fixing.

**Comment by ktoso** [comment]
A lot of GC and random timeouts unrelated to the changes...

**Comment by ktoso** [comment]
PLS BUILD

**Comment by ktoso** [comment]
Shall I merge this?

**Comment by patriknw** [comment]
yes, rebase and merge

**Comment by ktoso** [comment]
Only conflict was in migration guide, merging.

**Pull request merged**